src/TNode.h

```diff
@@ -0,0 +1,54 @@
+//
+// Created by shvmo on 29/04/2020.
+//
+
+#ifndef DATASTRUCTURESWET1_TNODE_H
+#define DATASTRUCTURESWET1_TNODE_H
+
+
+#include <memory>
+
+using std::shared_ptr;
+
+template<class Key, class Info>
+class TNode {
+public:
+    TNode();
+
+    ~TNode();
+
+    Key &getKey();
+
+    Info &getInfo();
+
+    shared_ptr<Info> getInfoPointer() const;
+
+    shared_ptr<TNode<Key, Info>> getLeft() const;
+
+    shared_ptr<TNode<Key, Info>> getRight() const;
+
+    void setKey(const Key &newKey);
+
+    int getHeight() const;
+
+    void setInfo(const Info &info);
+
+    void setInfoPointer(shared_ptr<Info> newInfoPointer);
+
+    void setLeft(shared_ptr<TNode<Key, Info>> left);
+
+    void setRight(shared_ptr<TNode<Key, Info>> right);
+
+    void setHeight(int height);
+
+private:
```

Comment:
why private after public? any special reason?

Reply:
That seemed to be the convention in every case we did projects in c++.
like here for example: https://github.com/Friziepop/mtm-ex3/blob/master/UniqueArray.h

Reply:
huh... didn't remember then... ok.